include/maths/analytics/CBoostedTreeUtils.h

```diff
@@ -50,37 +96,132 @@ enum EExtraColumnTag {
     E_BeginSplits
 };
 
-enum EHyperparameter {
-    E_DownsampleFactor = 0,
-    E_Alpha,
-    E_Lambda,
-    E_Gamma,
-    E_SoftTreeDepthLimit,
-    E_SoftTreeDepthTolerance,
-    E_Eta,
-    E_EtaGrowthRatePerTree,
-    E_MaximumNumberTrees, //!< Train only.
-    E_FeatureBagFraction,
-    E_PredictionChangeCost,     //!< Incremental train only.
-    E_RetrainedTreeEta,         //!< Incremental train only.
-    E_TreeTopologyChangePenalty //!< Incremental train only.
-};
-
 constexpr std::size_t NUMBER_EXTRA_COLUMNS{E_BeginSplits + 1}; // This must be last extra column
-constexpr std::size_t NUMBER_HYPERPARAMETERS{E_TreeTopologyChangePenalty + 1}; // This must be last hyperparameter
 constexpr std::size_t UNIT_ROW_WEIGHT_COLUMN{std::numeric_limits<std::size_t>::max()};
 
-//! \brief Hyperparameter importance information.
-struct MATHS_ANALYTICS_EXPORT SHyperparameterImportance {
-    enum EType { E_Double = 0, E_Uint64 };
-    EHyperparameter s_Hyperparameter;
-    double s_Value;
-    double s_AbsoluteImportance;
-    double s_RelativeImportance;
-    bool s_Supplied;
-    EType s_Type;
+//! \brief A fast ordered search tree.
+//!
+//! DESCRIPTION:\n
+//! This provides a single query upperBound(x), i.e. find the smallest value in
+//! an ordered set greater than the specified query point x. When possible it
+//! uses SSE-like instructions to perform a 4 way comparison between the query
+//! point and candidate split points. This means it achieves a branch factor of
+//! 5 and complexity O(ceil(log(n) / log(5))) in the set size n.
+//!
+//! IMPLEMENTATION DECISIONS:\n
```

Comment:
While you describe all important implementation decisions in words, I think it would be useful for posterity to document a specific example of input array and the tree representation.

Reply:
See [f690c42](https://github.com/elastic/ml-cpp/pull/2380/commits/f690c425b13a8eaa75f1c23b61f34191631249c4).